### ISSUE TITLE
Example how to work in browser

### ISSUE THREAD

**Issue by Enelar** [soliton4/promiseland#3]
How to setup code source map?
Should i use browserify?

**Comment by soliton4** [comment]
Hello @Enelar and sorry for my delayed reply.

i am very happy to see people are taking interest in promiseLand.

there is no sourcecode map implemented at the moment.
i am in the middle of rewriting the parser because of performance reasons. i was also planning to have a promiseland remote debugging tool which would be superior to in browser js debugging. but i guess its an ambitious project.

i will support source code map in the next minor release. 

if you want to go ahead and help with the implementation let me know. i can give you pointers.

we basically have all the original location of every expression / statement.
so it should not be to hard to implement.

**Comment by Enelar** [comment]
Yes, I'd love to, but first I need to get some experience. Could I use this
on client side? (like in browser)? Or it server only solution?

**Comment by soliton4** [comment]
the goal is to solve common shortages of javascript. so ofc you can use it in whatever javascript based runtime environment you want.

i recommend going through the features part of the wiki.

especially the frames part which shows you how you can easily connect server and browser code by using the same source base.
https://github.com/soliton4/promiseland/wiki/frames

https://www.npmjs.com/package/promiseland-webframework
is a reference implementation for a browser / node.js communication.
similar aproaches can be made to communicate between several node instances or workerthreads.

and if you want to get really deep into it there are synced objects which allow you to have the same instance of an object available accross several frames.

i am very interested in feedback from users. pls note that promiseLand is still in its early stage and will hopefully move forward a lot.

it also needs to keep up with new syntax models of ecmascript. since i want it to stay compatible with ecmascript.

**Comment by Enelar** [comment]
Sorry for delayed replay. Thanks a lot. I've got inspired with your work.
I've dreamed few years about pure async language, but had no idea how to build VM.
Your frames / inter-frame communication is genius.

**Comment by soliton4** [comment]
thank you for noticing ;)
the work on promiseland has kinda stagnated but i am working on a language that will contain most of the features, especially the synced objects and inter-frame function calls. it is not js specific but i want it to be able to translate to any possible language.
wanna get involved?
